### PR TITLE
Add timedelta support for applicable arguments of sensors

### PR DESCRIPTION
**Pull request by AlexisBRENON**
Related: #32512

### CODE REVIEW

File: airflow/sensors/base.py [hunk 1]
```diff
@@ -87,9 +87,18 @@ class BaseSensorOperator(BaseOperator, SkipMixin):
...
     :param soft_fail: Set to true to mark the task as SKIPPED on failure
-    :param poke_interval: Time in seconds that the job should wait in
-        between each try
-    :param timeout: Time, in seconds before the task times out and fails.
+    :param poke_interval: Time that the job should wait in between each try.
+        Can be ``timedelta`` or ``float`` seconds.
+    :param timeout: Time before the task times out and fails.
+        Can be ``timedelta`` or ``float`` seconds.
+        This should not be confused with ``execution_timeout`` of the
+        ``BaseOperator`` class. ``timeout`` measure the time between the
```
Comment (phanikumv) [inline airflow/sensors/base.py]:
```suggestion
        ``BaseOperator`` class. ``timeout`` measures the time elapsed between the
```

File: airflow/sensors/base.py [hunk 2]
```diff
@@ -87,9 +87,18 @@ class BaseSensorOperator(BaseOperator, SkipMixin):
     a criteria is met and fail if and when they time out.
 
     :param soft_fail: Set to true to mark the task as SKIPPED on failure
-    :param poke_interval: Time in seconds that the job should wait in
-        between each try
-    :param timeout: Time, in seconds before the task times out and fails.
+    :param poke_interval: Time that the job should wait in between each try.
+        Can be ``timedelta`` or ``float`` seconds.
+    :param timeout: Time before the task times out and fails.
```
Comment (phanikumv) [inline airflow/sensors/base.py]:
```suggestion
    :param timeout: Time elapsed before the task times out and fails.
```

File: airflow/sensors/base.py [hunk 3]
```diff
@@ -87,9 +87,18 @@ class BaseSensorOperator(BaseOperator, SkipMixin):
...
+        Can be ``timedelta`` or ``float`` seconds.
+        This should not be confused with ``execution_timeout`` of the
+        ``BaseOperator`` class. ``timeout`` measure the time between the
+        first poke and the current time (taking into account any
+        reschedule delay between each poke), while ``execution_timeout``
+        check the **running** time of the task (leaving out any reschedule
```
Comment (phanikumv) [inline airflow/sensors/base.py]:
```suggestion
        checks the **running** time of the task (leaving out any reschedule
```

**Comment by potiuk** [comment]
Nice! Thanks @AlexisBRENON for taking the time to improve docs. I likel the changes, I think though the title/description of the commit should be changed beceause it's more than doc change - it's also coertion and supporting timedelta.

**Comment by AlexisBRENON** [comment]
> Nice! Thanks @AlexisBRENON for taking the time to improve docs. I like the changes, I think though the title/description of the commit should be changed beceause it's more than doc change - it's also coertion and supporting timedelta.

Yeah, maybe it should be two different PRs ?
Or should I just update the title ?

**Review by potiuk — APPROVED** [review]

**Pull request merged**